Components/Core/Timer.cpp

```diff
@@ -1,34 +1,226 @@
+	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE) {
+		if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+			timerState = UNINITIALIZED;
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief Changes timer period, Sets timer state back to counting and starts timer
+ */
+bool Timer::ChangePeriodAndStart(const uint32_t period_ms)
+{
+	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE) {
+		timerState = COUNTING;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Starts the timer
+*/
+bool Timer::Start()
+{
+	if ((timerState == COMPLETE) || (timerState == COUNTING)) {
+		return false;
+	}
+	else if (timerState == PAUSED) {
+		ChangePeriod(remainingTimeBetweenPauses);
+	}
+	if (xTimerStart(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = COUNTING;
 		return true;
+	}
 	return false;
 }
+
+/**
+ * @brief Stops the timer
+*/
+bool Timer::Stop()
+{
+	// Checks if timer is in counting state because it cannot be stopped in any other state
+	if (timerState != COUNTING) {
+		return false;
+	}
+	// Calculates the time left on the timer before it is paused
+	remainingTimeBetweenPauses = rtosTimeRemaning();
+	if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = PAUSED;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Restarts timer without starting to count
+*/
+bool Timer::ResetTimer()
+{
+	if (timerState == UNINITIALIZED) {
+		return false;
+	}
+	if (xTimerReset(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+			timerState = UNINITIALIZED;
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief Restarts Timer and starts counting
+*/
+bool Timer::ResetTimerAndStart()
+{
+	if (timerState == UNINITIALIZED) {
+		return false;
+	}
+	if (xTimerReset(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = COUNTING;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Sets timer to auto-reload if parameter is set to true, Sets timer to one shot if parameter is set to false
+*/
+void Timer::SetAutoReload(bool setReloadOn)
+{
+	if (setReloadOn == true){
+		vTimerSetReloadMode(rtTimerHandle, pdTRUE);
+		//Testing purposes
+		SOAR_PRINT("Set to Auto Reload\n\n");
+	}
+	if (setReloadOn == false){
+		vTimerSetReloadMode(rtTimerHandle, pdFALSE);
+		//Testing purposes
+		SOAR_PRINT("Set to One Shot\n\n");
+	}
+}
+
+/**
+ * @brief Returns true if the timer is set to autoreload and false if it is set to one-shot
+*/
+bool Timer::GetAutoReload()
+{
+	if ((uxTimerGetReloadMode(rtTimerHandle)) == (( UBaseType_t ) pdTRUE)) {
+		return true;
+	}
+	if ((uxTimerGetReloadMode(rtTimerHandle)) == (( UBaseType_t ) pdFALSE)) {
```

Comment:
Can replace this second function call with an else, it can only return pdTRUE or pdFALSE.